packages/atclient/include/atclient/metadata.h

```diff
@@ -3,59 +3,172 @@
 #define ATCLIENT_METADATA_H
 
 #include "atclient/atstr.h"
+#include <stdbool.h>
+#include <stddef.h>
 
-#define DATE_STR_BUFFER_LENGTH 256 // can hold most date strings found in metadata
-#define GENERAL_BUFFER_LENGTH 8192 // can hold most things found in metadata
+#define DATE_STR_BUFFER_SIZE 256 // can hold most date strings found in metadata
+#define GENERAL_BUFFER_SIZE 8192 // can hold most metadata strings
 
 typedef struct atclient_atkey_metadata {
-  atclient_atstr createdby; // read and set by the protocol
-  atclient_atstr updatedby; // read and set by the protocol
-  atclient_atstr createdat; // date representing when key was created
-  atclient_atstr updatedat; // date representing when key was last modified
-  atclient_atstr status;
+  atclient_atstr createdby; // represents the atSign who created this atkey. read from protoocl string
```

Comment:
`protocol` typo

Reply:
Completely redid the documentation https://github.com/atsign-foundation/at_c/pull/117/commits/4e8571eceb5e5153954040a451df42337e82a4c6